lib/rabbitmq/ffi/ext/field_value.rb

```diff
@@ -40,7 +40,7 @@ def apply(value)
         when ::Symbol; [:utf8,  FieldValueValue.new(Bytes.from_s(value.to_s.encode!(Encoding::UTF_8)).pointer)]
         when ::Array;  [:array, FieldValueValue.new(Array.from_a(value).pointer)]
         when ::Hash;   [:table, FieldValueValue.new(Table.from(value).pointer)]
-        when ::Fixnum; [:i64,       (v=FieldValueValue.new; v[:i64]=value; v)]
+        when 1.class;  [:i64,       (v=FieldValueValue.new; v[:i64]=value; v)]
```

Comment:
It works for me on Ruby 2.2.2 - how old a version are you worried about?  If I recall correctly, `Integer` has been around quite a while as the superclass of `Fixnum` and `Bignum`.

```ruby
[1] pry(main)> case 1; when ::Integer; true; else false end
=> true
[2] pry(main)> RUBY_VERSION
=> "2.2.2
```